File: php/utility/FeatureHook.php
<?php
declare(strict_types=1);

// RealRest SDK utility: feature_hook

class RealRestFeatureHook
{
    public static function call(RealRestContext $ctx, string $name): void
    {
        if (!$ctx->client) {
            return;
        }
        $features = $ctx->client->features ?? null;
        if (!$features) {
            return;
        }
        foreach ($features as $f) {
            if (method_exists($f, $name)) {
                $f->$name($ctx);
            }
        }
    }
}

File: rb/utility/feature_hook.rb
# RealRest SDK utility: feature_hook
module RealRestUtilities
  FeatureHook = ->(ctx, name) {
    return unless ctx.client
    features = ctx.client.features
    return unless features
    features.each do |f|
      f.send(name, ctx) if f.respond_to?(name)
    end
  }
end
